AK/BuiltinWrappers.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2021, Nick Johnson <sylvyrfysh@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Concepts.h"
+#include "Platform.h"
+
+template<Unsigned IntType>
+inline constexpr int popcount(IntType value)
+{
+#if defined(__GNUC__) || defined(__clang__)
+    if constexpr (sizeof(IntType) <= 4)
+        return __builtin_popcount(value);
+    if constexpr (sizeof(IntType) == 8) {
+#    ifdef AK_OK_MACOS
+        return __builtin_popcountll(value);
+#    else
+        return __builtin_popcountl(value);
+#    endif
```

Comment:
I'm pretty sure `unsigned long` is only 32 bits wide on the i386 System V ABI, so we're truncating `u64` to 32 bits here. 

Why not just compare `sizeof(IntType)` to `sizeof(unsigned int)`, `sizeof(unsigned long)` and `sizeof(unsigned long long)` directly?

Reply:
Done, and working on i386!